src/Nitwork/ANitwork.cpp

```diff
@@ -0,0 +1,259 @@
+            startInputHandler();
+            startOutputHandler();
+            if (!startNitworkConfig(port, ip)) {
+                std::cerr << "Error: Nitwork config failed" << std::endl;
+                return false;
+            }
+            if (!startNitworkThreads(threadNb, tick)) {
+                std::cerr << "Error: Nitwork threads failed" << std::endl;
+                return false;
+            }
+            std::cout << "Nitwork started on port " << port << std::endl;
+        } catch (std::exception &e) {
+            std::cerr << "Nitwork Error : " << e.what() << std::endl;
+            return false;
+        }
+        return true;
+    }
+
+    bool ANitwork::startContextThreads(int threadNb)
+    {
+        for (int i = 0; i < threadNb; i++) {
+            _pool.emplace_back([this]() {
+                try {
+                    _context.run();
+                } catch (std::exception &e) {
+                    std::cerr << "Error: " << e.what() << std::endl;
+                }
+            });
+            if (!_pool.back().joinable()) {
+                std::cerr << "Error: thread nb: " << i << " not joinable" << std::endl;
+                return false;
+            }
+        }
+        return true;
+    }
+
+    bool ANitwork::startClockThread(int tick)
+    {
+        _clockThread = std::thread([this, tick]() {
+            try {
+                while (true) {
+                    std::this_thread::sleep_for(std::chrono::milliseconds(tick));
+                    std::unique_lock<std::mutex> lockTick(_tickMutex);
+                    _tickConvVar.notify_one();
+                }
+            } catch (std::exception &e) {
+                std::cerr << "Error: " << e.what() << std::endl;
+            }
+        });
+        if (!_clockThread.joinable()) {
+            std::cerr << "Error: clock thread not joinable" << std::endl;
+            return false;
+        }
+        return true;
+    }
+
+    bool ANitwork::startNitworkThreads(int threadNb, int tick)
+    {
+        if (!startContextThreads(threadNb)) {
+            std::cerr << "Error: context threads failed" << std::endl;
+            return false;
+        }
+        if (!startClockThread(tick)) {
+            std::cerr << "Error: clock thread failed" << std::endl;
+            return false;
+        }
+        return true;
+    }
+
+    void ANitwork::stop()
+    {
+        _context.stop();
+        for (auto &thread : _pool)
+            thread.join();
+        _pool.clear();
+        _clockThread.join();
+    }
+
+    void ANitwork::startReceiveHandler()
+    {
+        _receiveBuffer.fill(0);
+        _socket.async_receive_from(
+            boost::asio::buffer(_receiveBuffer),
+            _senderEndpoint,
+            boost::bind(
+                &ANitwork::headerHandler,
+                this,
+                boost::asio::placeholders::bytes_transferred,
+                boost::asio::placeholders::error));
+    }
+
```

Comment:
```suggestion
static void callReceiveHandler(std::string message)
{
    std::cerr << message << std::endl;
    startReceiveHandler();
}

```
To avoid some spaces to the below function